### DISCUSSION TITLE
Problem gelöst: Setzen von boolschen Variablen funktioniert in YAML mit 0/1

### DISCUSSION THREAD

**Discussion by oekinger** [SukramJ/aiohomematic#468] [General]
Hallo zusammen,

erst einmal muss ich mich für dieses großartige Projekt bedanken: Homematic macht gleich viel mehr Spaß :-).

Ich habe im MOment - auch mit der aktuellen Version 1.6.1 - leider ein komisches Problem beim Setzen von Variablen über die Funktion "Homematic(IP) Local: Set variable value":

- Parameter on wird korrekt übergeben -> Boolsche Variable in der CCU wechselt auf on
- Parameter off wird nicht korrekt übergeben -> Boolsche Variable in der CCU wechselt ebenfalls auf on bzw. wird mit on aktualisiert

Mein Setup: Hassio läuft auf einem X86 Mini-PC, RPI-RF-MOD ist mit einem HB-RF-USB 2 über einen USB-Hub direkt angeschlossen, Raspberrymatic läuft in der aktuellesten Version  3.63.9.20220521. Alle Homematic-Geräte (RF/IP) funktionieren einwandfrei, lesend haben alle Systemvariablen der CCU auch den korrekten Status. Nur beim Ändern des Status tritt das Verhalten auf, es gibt keinen Eintrag im Log (weder hassio noch auf der ccu).

Hat noch jemand dieses Verhalten oder habe ich einen spezifischen Fehler bei mir? Aufgefallen ist mir das erst nachdem einige Automationen nicht mehr korrekt gearbeitet haben und die Ursache ein falsche Status der CCU-Variablen war. Beholfen habe ich mir erst mal indem ich diese Funktion über Node Red erledige und aus meinen Automationen ausgebaut habe.

Danke schon mal und viele Grüße,
Marko

**Comment by SukramJ** [comment]
Aktuell funktioniert:
false --> 0
true --> 1

```
service: homematicip_local.set_variable_value
data:
  entity_id: homematicip_local.XXX
  name: XXX
  value: 0
```

**Reply by oekinger** [reply]
Ok, dann muss das spezifisch bei mir sein:

CCU ist 0 -> homematicip_local.set_variable_value 0 --> CCU ist 1 / on

Weiter getestet: egal welches Value ich übergebe (z. B. "rr"), die Variable in der CCU wechselt auf 1/true/on

Im Error-Log hab ich doch noch was dazu gefunden, ich kann es aber keiner Aktion zuordnen
[error.txt](https://github.com/danielperna84/hahomematic/files/8796394/error.txt)

**Reply by SukramJ** [reply]
WIE machst du denn was?
YAML oder UI?
Der zu übergebene Wert muss aktuell 0 oder 1 sein nicht "0" oder "1". Daher das YAML Beispiel.

**Reply by oekinger** [reply]
Jetzt verstehe ich: ich hab das über die GUI gemacht -> mit YAML funktioniert's.
Dankeschön, ich passe das gleich mal an 👍

**Reply by SukramJ** [reply]
Mit der 1.6.2 gibt es ein paar [Änderungen](https://github.com/danielperna84/custom_homematic#homematicip_localset_variable_value).

**Reply by oekinger** [reply]
Großartig, ich freue mich schon auf das Release. Danke für die tolle Arbeit!!

**Reply by oekinger** [reply]
Gerade die 1.6.2 angetestet: funktioniert nun perfekt auch über UI :-D
